openlibrary/plugins/upstream/addbook.py

```diff
@@ -530,6 +530,10 @@ def trim_doc(doc):
     return web.storage((k, trim_value(v)) for k, v in doc.items() if k[:1] not in "_{")
 
 
+class TocParseError(BaseException):
```

Comment:
Move this error into `table_of_contents.py` ; that is the code that "owns" it. That'll also prevent a circular dependency 😁